crates/sui-bridge/src/orchestrator.rs

```diff
@@ -131,31 +121,209 @@ where
     }
 
     async fn run_eth_watcher(
-        _store: Arc<BridgeOrchestratorTables>,
+        store: Arc<BridgeOrchestratorTables>,
         mut eth_events_rx: mysten_metrics::metered_channel::Receiver<(
             ethers::types::Address,
-            Vec<ethers::types::Log>,
+            Vec<EthLog>,
         )>,
-        _bridge_committee: Arc<ArcSwap<BridgeCommittee>>,
     ) {
         info!("Starting eth watcher task");
-        while let Some((_contract, logs)) = eth_events_rx.recv().await {
-            // TODO: skip events that are not already processed (in DB and on chain)
+        while let Some((contract, logs)) = eth_events_rx.recv().await {
+            if logs.is_empty() {
+                continue;
+            }
 
+            // TODO: skip events that are not already processed (in DB and on chain)
             let bridge_events = logs
                 .iter()
                 .map(EthBridgeEvent::try_from_eth_log)
                 .collect::<Vec<_>>();
 
+            let mut actions = vec![];
             for (log, opt_bridge_event) in logs.iter().zip(bridge_events) {
                 if opt_bridge_event.is_none() {
                     // TODO: we probably should not miss any events, warn for now.
                     warn!("Eth event not recognized: {:?}", log);
+                    continue;
```

Comment:
Why not start with `panic`?

Reply:
it's easier for testing. When everything is wired up I think panic is fine too.